### PR TITLE
MaxUnpooling2D typing

### PR DESCRIPTION
**Pull request by WindQAQ**
# Description

Silence typecheck warning.

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

Passes existing tests.

### CODE REVIEW

File: tensorflow_addons/layers/max_unpooling_2d.py [hunk 1]
```diff
@@ -105,7 +107,13 @@ class MaxUnpooling2D(tf.keras.layers.Layer):
     """
 
     @typechecked
-    def __init__(self, pool_size=(2, 2), strides=(2, 2), padding="SAME", **kwargs):
+    def __init__(
+        self,
+        pool_size: Union[list, tuple, int] = (2, 2),
```
Comment (thaink) [inline tensorflow_addons/layers/max_unpooling_2d.py:112]: Why do we need 'int' here?

Comment (WindQAQ) [reply tensorflow_addons/layers/max_unpooling_2d.py:112]: Pooling and conv layers also accept a single int to represent the same width and height, and `normalize_tuple` can deal with it.

Comment (thaink) [reply tensorflow_addons/layers/max_unpooling_2d.py:112]: I see. Thanks.

**Review by thaink — APPROVED** [review]

**Review by seanpmorgan — APPROVED** [review]
LGTM thanks!

**Comment by seanpmorgan** [comment]
And thanks @thaink for the review!
